src/frontend/components/VideoPlayer/index.tsx

```diff
@@ -175,3 +178,41 @@ export class VideoPlayer extends React.Component<
     );
   }
 }
+
+export type VideoPlayerConnectedProps = Pick<
+  VideoPlayerProps,
+  'createPlayer' | 'video'
+>;
+
+/**
+ * Replace the (read-only) video from context with one from the resources part of the
+ * state if available as it will hold the most recent version.
+ */
+export const mapStateToProps = (
+  state: RootState<appStateSuccess>,
+  { video }: VideoPlayerConnectedProps,
+) => ({
+  jwt: state.context.jwt,
+  languageChoices: state.languageChoices.items,
+  thumbnail: getThumbnail(state),
+  timedtexttracks: getTimedTextTracks(state),
+  video:
+    (state.resources[modelName.VIDEOS]!.byId &&
+      state.resources[modelName.VIDEOS]!.byId[(video && video.id) || '']) ||
+    video,
+});
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  dispatch,
+  getTimedTextTrackLanguageChoices: (jwt: string) =>
+    dispatch(getTimedTextTrackLanguageChoicesAction(jwt)),
+});
+
+/**
+ * Component. Displays a player to show the video from context.
+ * @param video The video to show.
+ */
+export default connect(
```

Comment:
I think we still want named imports. This should be `export const VideoPlayer = connect...`